File: performance_tests-py2many/sieve/sieve_numpy.jl

function sieve(n)
    primes = ones(Float64, n)
    (primes[1], primes[2]) = (false, false)
    for i = 2:Int(sqrt(n) + 1)-1
        if primes[i+1]
            primes[end:i:i*i+1] = false
        end
    end
    return [i for i in primes if primes[i] != 0]
end

if abspath(PROGRAM_FILE) == @__FILE__
    sieve(parse(Int, append!([PROGRAM_FILE], ARGS)[2]))
end

File: user_tests/external_modules/sieve_numpy.py
import numpy as np


def sieve(n):
    primes = np.ones(n, dtype=bool)
    primes[0], primes[1] = False, False
    for i in range(2, int(np.sqrt(n) + 1)):
        if primes[i]:
            # Vectorization of the assignment operation (Uses a loop in C)
            primes[i*i::i] = False 
    return np.flatnonzero(primes)